Adds services payment article (#155)
moondance-labs/tanssi-docs: learn/tanssi/services-payment.md

---

> title: Services Payment

Reviewer: ```suggestion
title: Block Production Services Payment
```

For now, let's add Block Production to the beginning of the title so it's a bit more descriptive in search engine results

> description: Appchains deployed through Tanssi benefit from block production services provided by a set of node operators, which are compensated with Tanssi tokens.

Reviewer 2: ```suggestion
description: Appchains deployed through Tanssi benefit from block production services provided by a set of node operators, who are compensated through staking rewards.
```

---

# Services Payments {: #technical-features-of-tanssi }

## Introduction {: #introduction }

As presented in the [Overview](/learn/tanssi/overview/){target=\_blank} article, Tanssi is an infrastructure protocol that addresses the complexities and high costs associated with setting up and maintaining blockchain infrastructure, streamlining the deployment of appchains. This protocol brings benefits for both participants:

- **Appchains** - teams can focus on the core logic of their product, the UX, and the UI without dealing with the challenges of infrastructure bootstrapping and its management

> - **Node operators** - bearing with the responsibility of keeping their hardware and software configuration in optimal conditions, they are rewarded with Tanssi tokens for executing transactions and producing blocks on behalf of Tanssi and Tanssi appchains

Reviewer 2: ```suggestion
- **Node operators** - bearing with the responsibility of keeping their hardware and software configuration in optimal conditions, they are incentivized for executing transactions and producing blocks on behalf of Tanssi and Tanssi appchains
```

> The [block production as a service](/learn/tanssi/technical-features/#block-production-as-a-service){target=\_blank} mechanism randomly selects a subset of a decentralized set of nodes and assigns them to provide services to Tanssi and the Tanssi-powered appchains for a limited period of time (session). Once the session ends, the Tanssi protocol will again select the set of block producers needed to serve Tanssi and all the active Tanssi appchains and re-assign them.

Reviewer 2: ```suggestion
The [block production as a service](/learn/tanssi/technical-features/#block-production-as-a-service){target=\_blank} mechanism randomly selects a subset of a decentralized set of nodes and assigns them to provide services to Tanssi and the Tanssi-powered appchains for a limited period, that is, the next session (a set number of blocks). Once an ongoing session ends, the Tanssi protocol rotates the set of block producers needed to serve Tanssi and all the active Tanssi appchains, and they automatically start serving the chain they were assigned to.
```

All this operation carries costs that must be covered by the appchains that want to obtain the services. This article covers the general aspects of those costs and associated service payments.

## Costs to Cover  {: #costs-to-cover }

There are three main costs associated with block production as a service that any appchain must cover using Tanssi tokens to deploy successfully and get the block production services:

- **Registration deposit** - is the initial deposit that is locked from the account that signs the appchain registration transaction

> - **Block producers assignment** - every time the Tanssi protocol assigns block producers, which happens once per session, a fixed fee is charged. This discourages a faulty appchain (whose runtime logic fails to produce valid transitions and, therefore, a valid block) from requesting being assigned nodes that will not be able to produce valid blocks

Reviewer: ```suggestion
- **Block producers assignment** - every time the Tanssi protocol assigns block producers, which happens once per session, a fixed fee is charged. This discourages a faulty appchain (one whose runtime logic fails to produce valid transitions and, therefore, a valid block) that cannot generate valid blocks from requesting node assignments 
```

> - **Block producers assignment** - every time the Tanssi protocol assigns block producers, which happens once per session, a fixed fee is charged. This discourages a faulty appchain (whose runtime logic fails to produce valid transitions and, therefore, a valid block) from requesting being assigned nodes that will not be able to produce valid blocks

Reviewer 2: I don't like "faulty" here, normally we would use "malicious"?

Author: Maybe keep both "faulty or malicious"?
The incentives would be: "fix your chain" in one case and "messing with tanssi is expensive" in the other

Reviewer 2: Maybe leave it to "This fee gives appchains the right to be assigned block producers, and should discourage appchains whose runtime logic fails to produce valid transactions or blocks."

- **Block production** - appchains need to pay for each block that is produced on their behalf. Since the protocol selects and assigns the block producers on a per-session basis, appchains must have enough funds to cover all the blocks to be produced in an entire session to be served

> On Dancebox, the Tanssi testnet, the current configuration is the following:

Reviewer: ```suggestion
On Dancebox, the Tanssi TestNet, the current configuration is the following:
```

> On Dancebox, the Tanssi testnet, the current configuration is the following:

Reviewer 2: ```suggestion
On Dancebox, the Tanssi TestNet, the current configuration is the following:
```

I suggest removing this as this is short-lived. Once Tanssi is in Mainnet we just add tabs to the table. Maybe changing it to "The current configuration is set as follows:"

=== "Dancebox"
    |         Variable          |                                   Value                                   |
    |:-------------------------:|:-------------------------------------------------------------------------:|
    |   Registration deposit    |         {{ networks.dancebox.costs.registration_deposit }} DANCE          |

>     | Block producer assignment | {{ networks.dancebox.costs.cost_per_assignment }} microDANCE per session |

Reviewer 2: We never discuss what is a "microDance" so I would just put units in  XYZ x 10^-6 or something?

    |     Block production      |    {{ networks.dancebox.costs.cost_per_block }} microDANCE per block     |

> To ensure block production in the next session, the total balance must be at least enough to cover the block producer assignment cost plus the cost to produce the {{ networks.dancebox.session.blocks }} blocks that comprise an entire session.

Reviewer: how is this paid for? i'm assuming in the dapp?

Author: Yup, in the dapp there's a "top-up" section (the tipping mechanism is not yet there though)

!!! note
    Although these costs are currently fixed, as protocol development progresses, they might become dynamic, varying in response to the network's workload.

## Tipping {: #tipping }

> Besides the regular costs presented in the previous section, there is an additional and optional fee: the tipping mechanism. Appchains are free to offer a tip to be served with priority over the rest, which, in times when the network is experiencing a high demand that can not be met with the available resources, could be the difference between producing blocks or stalling for an entire session at least.  

Reviewer 2: I would switch this paragraph a bit so that we state "The Problem -> The Solution"

So you start like:

"On some occasions, Tanssi might experience high demand for its block production services that can not be met with the available resources. For example, if for the next session, there are ten appchains and Tanssi can only serve eight, two appchains might stall for the entire session duration...."

> As explained in the [block producers](/block-producers/onboarding/account-setup/#verify){target=\_blank} section, when there are more available block producers than the required to provision Tanssi and all the Tanssi appchains, then those who have more tokens staked are the ones selected to participate in the block production for the next session, leaving those with fewer funds staked on hold. Similarly, when there are more appchains to serve than block producers to assign, the appchains that can be served with the available block producers will be selected according to the priority given by the tips they offer.

Reviewer: I like the idea of a comparison, but maybe comparing it to Ethereum priority fees would make more sense for some stakeholders that are reading this particular guide. What do you think?
